src/snowflake/snowpark/dataframe_transform_functions.py

```diff
@@ -120,3 +122,85 @@ def moving_agg(
                     agg_df = agg_df.with_column(formatted_col_name, agg_col)
 
         return agg_df
+
+    def compute_lag(
```

Comment:
looks like compute_lag and compute_lead shares a lot common code? can we extract the common part?

Reply:
done.